### PR TITLE
add backup and update for preds caches

### PR DESCRIPTION
**Pull request by tybalex**
**What does this PR do:**
backup the cached nulog predictions to a local file and sync up the file to minio.

**Any Dependency on Other works/PRs:** No

### CODE REVIEW

File: nulog-inference-service/start-nulog-inference.py [hunk 1]
```diff
@@ -28,8 +30,18 @@
 ES_ENDPOINT = os.environ["ES_ENDPOINT"]
 ES_USERNAME = os.getenv("ES_USERNAME", "admin")
 ES_PASSWORD = os.getenv("ES_PASSWORD", "admin")
+MINIO_ACCESS_KEY = os.environ["MINIO_ACCESS_KEY"]
```
Comment (AmartC) [inline nulog-inference-service/start-nulog-inference.py]: These environment variables will need to change as we are switching over to more general S3 storage and not specifically Minio as introduced in this PR: https://github.com/rancher/opni-inference-service/pull/8

**Review by AmartC — CHANGES_REQUESTED** [review]

File: nulog-inference-service/start-nulog-inference.py (hunk 1 above)
Comment (tybalex) [reply nulog-inference-service/start-nulog-inference.py]: I will merge from that PR once it's merged.

**Review by sanjay920 — DISMISSED** [review]
LGTM
I did notice that you use `logger.debug` instead of `info` - why?

File: nulog-inference-service/start-nulog-inference.py (hunk 1 above)
Comment (tybalex) [reply nulog-inference-service/start-nulog-inference.py]: merged from main, pls review again

**Comment by tybalex** [comment]
> LGTM
> I did notice that you use `logger.debug` instead of `info` - why?

for some logger, we may only want to see them when debugging.

**Review by AmartC — DISMISSED** [review]

File: nulog-inference-service/start-nulog-inference.py [hunk 2]
```diff
@@ -111,11 +132,77 @@ async def doc_generator(df):
         logger.error(e)
 
 
+def s3_setup(s3_client):
+    # Function to set up a S3 bucket if it does not already exist.
+    try:
+        s3_client.meta.client.head_bucket(Bucket=S3_BUCKET)
+        logger.debug("{S3_BUCKET} bucket exists")
```
Comment (AmartC) [inline nulog-inference-service/start-nulog-inference.py]: Needs to be an f-string. Everything else looks good.

**Review by AmartC — CHANGES_REQUESTED** [review]
Just one minor fix. Everything else looks good.

File: nulog-inference-service/start-nulog-inference.py (hunk 2 above)
Comment (tybalex) [reply nulog-inference-service/start-nulog-inference.py]: patched.

**Review by AmartC — APPROVED** [review]
LGTM

**Review by sanjay920 — APPROVED** [review]

**Pull request merged**